packages/babel-core/src/transformation/file/index.js

```diff
@@ -54,6 +54,7 @@ export default class File extends Store {
       sourceType:    this.opts.sourceType,
       filename:      this.opts.filename,
       sourceFileName:this.opts.filename,
+      allowReturnOutsideFunction: true,
```

Comment:
The lets wait for #3561. I wouldn't add that now, as removing it again would be a breaking change if we are careful.